### DISCUSSION TITLE
deploy FastAPI app on windows server with IIS and FastCGI not working

### DISCUSSION THREAD

**Discussion by GitTeamIT** [fastapi/fastapi#11799] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
    <system.webServer>
        <handlers>
            <add name="botWhatsapp" path="*" verb="*" modules="FastCgiModule" scriptProcessor="C:\Users\Administrator\AppData\Local\Programs\Python\Python311\python.exe|C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\site-packages\wfastcgi.py" resourceType="Unspecified" requireAccess="Script" />
        </handlers>
    </system.webServer>
    <appSettings>
        <add key="PYTHONPATH" value="C:\inetpub\wwwroot\bot.whatsapp" />
        <add key="WSGI_HANDLER" value="main.app" />
    </appSettings>
</configuration>
        </handlers>
    </system.webServer>
    <appSettings>
        <add key="PYTHONPATH" value="C:\inetpub\wwwroot\bot.whatsapp"/>
        <add key="WSGI_HANDLER" value="main.app"/>
    </appSettings>
</configuration>
```

### Description

I wanna deploy my fastAPI app on a windows server with IIS and FastCGI. I already configured the handler mapping and the web site on IIS Manager, but it throws me and 500 error

![image](https://github.com/tiangolo/fastapi/assets/173852751/8364dd8e-1935-4e42-8206-b5b3691ccce8)

In the example code i put the web.config. Any idea of what i'm doing worng? Or is there another method to deploy the application on windows server?

### Operating System

Windows

### Operating System Details

Windows Server 2022

### FastAPI Version

0.111.0

### Pydantic Version

2.8.0po

### Python Version

Python 3.11.9

### Additional Context

Fast CGI settings:
![image](https://github.com/tiangolo/fastapi/assets/173852751/00279cc2-0154-490a-b740-79867e5fdfcf)

Handler mapping:
![image](https://github.com/tiangolo/fastapi/assets/173852751/7967f746-0872-432e-8b43-04dfe6a71cc5)

Executable for Handler mapping:
C:\Users\Administrator\AppData\Local\Programs\Python\Python311\python.exe|C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\site-packages\wfastcgi.py

web site details
![image](https://github.com/tiangolo/fastapi/assets/173852751/b1dd78a3-664d-4c83-95e0-946db3b27165)

**Comment by GitTeamIT** [comment]
```
<?xml version="1.0" encoding="UTF-8"?>
<configuration>
    <system.webServer>
        <handlers accessPolicy="Read, Execute, Script">
            <add name="botWhatsapp" path="*" verb="*" modules="FastCgiModule" scriptProcessor="C:\Users\Administrator\AppData\Local\Programs\Python\Python311\python.exe|C:\Users\Administrator\AppData\Local\Programs\Python\Python311\Lib\site-packages\wfastcgi.py" resourceType="Unspecified" requireAccess="Script" />
        </handlers>
    </system.webServer>
    <appSettings>
        <add key="PYTHONPATH" value="C:\inetpub\wwwroot\bot.whatsapp" />
        <add key="WSGI_HANDLER" value="main.app" />
    </appSettings>
</configuration>
```

This is the actual web.config. I accidentaly duplicated the appsettings in the example code

**Comment by jalwn** [comment]
were you able to figure this out. i have the same issue as well
